src/en/guide/basics/tasks.md

```diff
@@ -35,3 +36,20 @@ async def explicit_inject(app):
 app.add_task(explicit_inject(app))
 ```
 :---
+
+## Adding tasks before `app.run`
+
+It is possible to add background tasks before the App is run ie. before `app.run`. To add a task before the App is run, it is recommended to not pass the coroutine object (ie. one created by calling the `async` callable), but instead just pass the callable and Sanic will create the coroutine object on **each worker**. Note: the tasks that are added such are run as `before_server_start` jobs and thus run on every worker (and not in the main process). This has certain consequences, please read [this comment](https://github.com/sanic-org/sanic/issues/2139#issuecomment-868993668) on [this issue](https://github.com/sanic-org/sanic/issues/2139) for further details.
+
+---:1
+
+To add a task before `app.run`
+:---:1
+```python
+async def slow_work(...):
+   ...
+
+app = Sanic(...)
+app.add_task(slow_work) # Note: we are passing the callable and not coroutine object `slow_work(...)`
+app.run(...)
+```
```

Comment:
Do you think we should add a "tip" here about `partial()`?

Also, since we are being explicit here about running on every work v the main process (which is of course accurate), I wonder if it is worth mentioning `@app.main_process_start` with a link to `./listeners.md`.

FYI, the syntax for "tips":

```
::: tip
info here
:::
```

Reply:
Sure.. Will Add.